cmd/api/src/api/v2/saved_queries.go

```diff
@@ -178,15 +178,41 @@ func (s Resources) DeleteSavedQuery(response http.ResponseWriter, request *http.
 	} else if err != nil {
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request), response)
 	} else if !savedQueryBelongsToUser {
-		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "invalid saved_query_id supplied", request), response)
-	} else if err := s.DB.DeleteSavedQuery(request.Context(), savedQueryID); errors.Is(err, database.ErrNotFound) {
-		// This is an edge case and can only occur if the database has a concurrent operation that deletes the saved query
-		// after the check at s.DB.SavedQueryBelongsToUser but before getting here.
-		// Still, adding in the same check for good measure.
-		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusNotFound, "query does not exist", request), response)
-	} else if err != nil {
-		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request), response)
+		_, isAdmin := user.Roles.FindByName(auth.RoleAdministrator)
+		if publicQueries, err := s.DB.GetPublicSavedQueries(request.Context()); err != nil {
+			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request), response)
+		} else {
+			isPublic := false
+			for _, query := range publicQueries {
+				if query.ID == int64(savedQueryID) {
+					isPublic = true
+					break
+				}
+			}
```

Comment:
This portion should probably use a new database call. Looping through all public queries is likely to have negative performance impacts as an instance gets more public queries, and doing this logic at the handler level isn't ideal.

We could write a new function like `db.IsSavedQueryPublic(context.Context, queryID)` and pull that specific query to see if it's marked public or not.

Reply:
add db call added